ui/components/app/signature-request/signature-request.component.js

```diff
@@ -45,9 +45,72 @@ export default class SignatureRequest extends PureComponent {
       hardwareWalletRequiresConnection,
     } = this.props;
     const { address: fromAddress } = fromAccount;
-    const { message, domain = {} } = JSON.parse(data);
+    const { message, domain = {}, types, primaryType } = JSON.parse(data);
     const { metricsEvent } = this.context;
 
```

Comment:
It will be easy to understand `sanitizeMessage` if it can be extracted into a util function and test covered.